Unity-Technologies/ScriptableRenderPipeline
# Shadergraph/validation refactor

### Checklist for PR maker
---
### Purpose of this PR
First step of Graph validation. This step was just pulling out everything that GraphData.ValidateGraph() does and trying to give clearer function calls

---
### Testing status

**Manual Tests**: What did you do?
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
See https://github.com/Unity-Technologies/ScriptableRenderPipeline/tree/shadergraph/first-pass-integration-tests/com.unity.shadergraph

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.
